### PR TITLE
Add some mimetypes

### PR DESCRIPTION
**Pull request by farukuzun**
For [owncloud/richdocuments](https://github.com/owncloud/richdocuments) to work correctly with following file formats;
- `application/vnd.lotus-wordpro`
- `application/vnd.visio`
- `application/vnd.wordperfect`
- `application/msonenote`

### CODE REVIEW

**Comment by PVince81** [comment]
@rullzer

File: core/js/mimetypelist.js [hunk 1]
```diff
@@ -82,26 +86,26 @@ OC.MimeTypeList={
...
+    "folder-starred",
+    "folder-public",
+    "folder-drag-accept",
+    "text",
+    "video",
+    "audio",
+    "file",
+    "application-pdf",
+    "folder-external",
+    "application"
```
Comment (nickvergessen) [inline core/js/mimetypelist.js]: if we toucht hte list like that, we should sort it alphabetically?

Comment (farukuzun) [inline core/js/mimetypelist.js]: @nickvergessen this file updated automatically by ./occ maintenance:mimetype:update-js command.

Comment (nickvergessen) [inline core/js/mimetypelist.js]: Okay, fixed it in https://github.com/owncloud/core/pull/21769

Comment (farukuzun) [inline core/js/mimetypelist.js]: Ah I didn't pay attention to this file since its automatically created, sorry for that :)

Comment (nickvergessen) [inline core/js/mimetypelist.js]: Merged the sorting, now you need to rebase your branch on master:
> This branch has conflicts that must be resolved

**Comment by farukuzun** [comment]
@nickvergessen Rebased

**Comment by rullzer** [comment]
We probabaly want a mimetype repair job for this as well. Else only new files have the proper mime types.

@farukuzun see https://github.com/owncloud/core/blob/master/lib/private/repair/repairmimetypes.php . The file is pretty self explaining. And of course unit tests in https://github.com/owncloud/core/blob/master/tests/lib/repair/repairmimetypes.php

**Comment by LukasReschke** [comment]
Missing unit tests, moving back to 2. @farukuzun let us know if you need any help on the unit tests and feel free to join us on #owncloud-dev in Freenode :smile: :rocket:

**Comment by farukuzun** [comment]
Thank you for your help and pardon my timing. I updated my commit. On [CODE](https://www.collaboraoffice.com/code/), richdocuments is working on ownCloud 8.2.2. Is it possible for me to backport this patch to next bugfix release for ownCloud 8? Thanks ^^

File: lib/private/repair/repairmimetypes.php [hunk 2]
```diff
@@ -356,5 +367,11 @@ public function run() {
 				$this->emit('\OC\Repair', 'info', array('Fixed rtf mime type'));
 			}
 		}
+
+		if (version_compare($ocVersionFromBeforeUpdate, '9.0.0.7', '<')) {
```
Comment (nickvergessen) [inline lib/private/repair/repairmimetypes.php]: Should use `9.0.0.9` and update the version in [version.php](https://github.com/owncloud/core/blob/master/version.php#L28)

Comment (farukuzun) [inline lib/private/repair/repairmimetypes.php]: Updated my commit.

**Comment by rullzer** [comment]
Looking good... one more thing...  could you update https://github.com/farukuzun/core-1/blob/master/tests/lib/repair/repairmimetypes.php#L449 as well?

**Comment by rullzer** [comment]
Also @farukuzun did you sign the CLA?

**Comment by MorrisJobke** [comment]
> Also @farukuzun did you sign the CLA?

Or as alternative simply state that this is MIT licensed ;)

**Comment by MorrisJobke** [comment]
Could I also ask you to rebase and change the version number to 10, because it was increased recently to 9.

**Comment by farukuzun** [comment]
Rebased, added changes @rullzer & @MorrisJobke asks, signed CLA and mailed it to @karlitschek

**Comment by karlitschek** [comment]
I can confirm that I got the agreement. :+1:

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Pull request merged**